js_modules/dagit/packages/ui/src/components/calculateMiddleTruncation.tsx

```diff
@@ -0,0 +1,31 @@
+/**
+ * Binary search to find the maximum middle-truncated text that will fit within the specified target width.
+ * The truncation will occur in the center of the string, with the same number of characters on either side.
+ */
+export const calculateMiddleTruncation = (
+  text: string,
+  targetWidth: number,
+  measure: (value: string) => number,
+): string => {
+  // Skip the search if the text will already fit as-is.
+  if (measure(text) <= targetWidth) {
+    return text;
+  }
```

Comment:
This is also new. Go ahead and measure the given text up front, and skip the truncation if we don't need it.